docs/DeveloperGuide.md

```diff
@@ -1,34 +1,579 @@
+
+This section contains links to other relevant guides.
+
+* [Documentation guide](./Documentation.md)
+* [Testing guide](./Testing.md)
+* [DevOps guide](./DevOps.md)
+
+-----------------------------------------------------------------------------------------------------------------------
+
+## **Implementation**
+
+This section describes some noteworthy details on how certain features are implemented.
+
+### Add task feature
+The adding of task mechanism is facilitated by `LogicManager`.
+
+The retrieving of task details is done by `Parser`. It splits the user input based on the command type and the various parameters' description.
+The user can input the parameters in any order.
+The `AddCommand` class will then be executed and the task will be added to the `TaskList`.
+
+`Parser`, `AddCommand`, `TaskList` implements the following operations:
+
+* `Parser#parse` - Parse user input to retrieve the command type and task details.
+* `Parser#parseAddCommand` - Parse user input to retrieve the respective parameters' details and creates an `AddCommand` object.
+* `AddCommand#execute` - Add task into `TaskList` and calls `Ui` to print message output.
+* `TaskList#addTask` - Create a task based on the given parameters and add it into the list.
+* `Storage#saveTaskListData` - Writes the current task list into the save file.
+
+Given below is an example usage scenario and how the task adding mechanism behaves at each step.
+
+**Step 1.** The user launches the application for the first time. The `TaskList` is initialized to be empty.
+
+![AddTaskEmptyListObjectDiagram](objectDiagrams/addTask/empty.png)
+
+**Step 2.** The user adds a task to the application, by entering `add n/Assignment1 t/1100 D/16-09-2020 d/2 r/Today i/high a/Refer to lecture notes`. The input will be read in by the `Athena` class. The input is passed into `LogicManager` which calls `Parser#parse` to parse the user input through `Parser#parseAddCommand`, to create an `AddCommand` object. The `AddCommand` object is returned to the `LogicManager`.
+
+**Step 3.** The `LogicManager` calls `AddCommand#execute`, which calls `TaskList#addTask` to create a task based on the given parameters, and adds the task to the list. The `TaskList` now contains 1 task (Assignment1). The message to show if the task is added successfully is subsequently outputted by the `AthenaUi` class to the user.
+
+![AddTaskEmptyListObjectDiagram](objectDiagrams/addTask/onetask.png)
+
+**Step 4.** After the command is executed, `LogicManager` calls `Storage#saveTaskListData` to automatically save the tasks in the `TaskList` into the save file.
+
+**Step 5.** Upon completion of execution, `LogicManager` returns a boolean value `false` to `Athena` to allow the continuous run of the program. 
+
+The following sequence diagram illustrates how Step 3 of the task adding operation works:
+
+![AddTaskSequenceDiagram](sequenceDiagrams/AddCommand%20diagram.png)
```

Comment:
Should the headbox be :AddCommand and :TaskList with underline instead? and the footbox should be removed?
<img width="492" alt="Screenshot 2020-10-30 at 14 21 12" src="https://user-images.githubusercontent.com/56800018/97667667-5dcf3800-1abb-11eb-8e5c-500b46520675.png">